### PR TITLE
Search bar component

### PR DESCRIPTION
**Pull request by aman-rana-vs**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
-# react-native-vs-animated-component
\ No newline at end of file
+# react-native-vs-animated-component
+
+# Search Bar
```
Comment (praveen-verma-vs) [inline README.md:3]: Add gif to preview the component ui

File: AnimatedComponents/src/components/search-bar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  /** iconColorExpanded: is an optional prop which dictates the color of the icon when search bar is expanded */
+  iconColorExpanded?: string
+  /** iconSize: is an optional prop which dictates the size of the icon */
+  iconSize?: number
+  /** iconStyle: is an optional prop which dictates the style of the search icon */
+  iconStyle?: DefaultStyle
+  /** inputContainerStyle: is an optional prop which dictates the style of the input container */
+  inputContainerStyle?: DefaultStyle
+  /** onInputChange: is a prop which handles the input change */
+  onInputChange: (text: string) => void
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/search-bar/SearchBar.tsx:32]: name it to handleInputChange as its a handler function.

File: AnimatedComponents/src/components/search-bar/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          </Animated.View>
+        )}
+        <TouchableOpacity
+          onPress={handlePress}
+          style={[
+            styles.searchIconContainer,
+            { backgroundColor: showInput ? iconColorCollapsed : iconColorExpanded },
+            iconStyle,
+          ]}>
+          {icon ? (
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/search-bar/SearchBar.tsx:108]: we can write it this way:
icon??<Icon ...>

File: AnimatedComponents/src/components/search-bar/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+  const [showInput, setShowInput] = useState(false)
+  const width = useSharedValue(50)
+  const containerWidth = useRef(0)
+
+  const handlePress = () => {
+    const newShowInput = !showInput
+    setShowInput(newShowInput)
+
+    width.value = withTiming(newShowInput ? containerWidth.current * 1 : 50, {
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/search-bar/SearchBar.tsx:67]: Use translate property instead of animating width.

**Review by pvn013 — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,51 @@
...
+
+The Search Bar component is an expandable search bar component which can expand and collapse by clicking on search icon.
+
```
Comment (praveen-verma-vs) [inline README.md:6]: Add "Preview" heading

File: AnimatedComponents/App.tsx [hunk 6]
```diff
@@ -67,6 +69,7 @@ const App = () => {
 
   return (
     <SafeAreaView style={{ flex: 1 }}>
+      <SearchBarDemo />
```
Comment (praveen-verma-vs) [inline AnimatedComponents/App.tsx:72]: Add title before the component and a sepration line at the bottom as well.

File: AnimatedComponents/src/components/search-bar/SearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  searchIconBackgroundColor?: string
+  /** searchIconColor: is an optional prop which indicates the color of the search icon */
+  searchIconColor?: string
+  /** searchValue: is a prop which holds the value of the search bar */
+  searchValue: string
+  /** wrapperContainerStyle: is an optional prop which holds the style of the main container which wraps the search bar */
+  wrapperContainerStyle?: DefaultStyle
+}
+
+const SearchBar = (props: ISearchBarProps) => {
```
Comment (praveen-verma-vs) [inline AnimatedComponents/src/components/search-bar/SearchBar.tsx:50]: Do the requested changes in search bar animation.

**Review by pvn013 — CHANGES_REQUESTED** [review]

**Comment by pvn013** [comment]
Update the video as well for the recent implementation.

**Pull request merged**